src/integer_value.cpp

```diff
@@ -116,27 +117,26 @@ ValuePtr IntegerValue::mul(Env *env, ValuePtr arg) {
     }
     arg.assert_type(env, Value::Type::Integer, "Integer");
 
+    if (is_zero() || arg->as_integer()->is_zero())
```

Comment:
```suggestion
    if (is_zero() || arg->to_nat_int_t() == 0)
```

Avoiding a call to hydrate().

Reply:
Well, you either amended the commit, or I had a really outdated page in my browser. This looks resolved.